exercises/diffie-hellman/tests/diffie-hellman.rs

```diff
@@ -0,0 +1,60 @@
+extern crate diffie_hellman;
+
+use diffie_hellman::*;
+
+#[test]
+fn test_private_key_in_range_key() {
+    let primes: Vec<u64> = vec![5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 773, 967,
+                                3461, 6131];
+    let private_keys: Vec<u64> = primes.iter().map(|x| private_key(*x)).collect();
+
+    for i in 0..primes.len() {
+        assert!(1 <= private_keys[i] && private_keys[i] < primes[i]);
```

Comment:
> Hmm I think then the [description](https://github.com/hekrause/problem-specifications/blob/master/exercises/diffie-hellman/description.md) is wrong.
> I think in this section of the [wiki](https://en.wikipedia.org/wiki/Diffie%E2%80%93Hellman_key_exchange) they are taking ```a``` from a pool of known numbers.
We are working here with primes where 1 is no prime and therefore should not be available in the range.

I understand. In that case I have changed my mind, let us not allow 1.

the reason I do not allow 1 as a private key is not because it is composite (otherwise, we would also have to disallow all other composite numbers, and this line does no such thing), but rather because the description says so, and perhaps it is for because 1 seems like a weak key (identity in exponentiation)